sdgym/collect.py

```diff
@@ -16,20 +19,47 @@ def collect_results(input_path, output_file):
             If ``output_file`` is provided, the consolidated
             results will be written there. Otherwise, they
             will be written to ``input_path``/results.csv.
+        aws_key (str):
+            If an ``aws_key`` is provided, the given access
+            key id will be used to read from and/or write to
+            any s3 paths.
+        aws_secret (str):
+            If an ``aws_secret`` is provided, the given secret
+            access key will be used to read from and/or write to
+            any s3 paths.
     """
     print(f'Reading results from {input_path}')
     run_path = pathlib.Path(input_path)
 
     scores = []
-    for path in tqdm.tqdm(list(run_path.glob('**/*.csv'))):
-        scores.append(pd.read_csv(path))
+    if is_s3_path(input_path):
```

Comment:
Just like the `write_file` function, I would move this to the `s3` module as a `read_file` function (and maybe also `read_csv`) that is able to decide between reading from S3 and reading from a local path.